x/merkledb/proof_test.go

```diff
@@ -839,68 +835,16 @@ func Test_ChangeProof_Syntactic_Verify(t *testing.T) {
 			expectedErr: ErrStartAfterEnd,
 		},
 		{
-			name: "no roots in history and non-empty key-values",
-			proof: &ChangeProof{
-				HadRootsInHistory: false,
-				KeyChanges:        []KeyChange{{Key: []byte{1}, Value: Some([]byte{1})}},
-			},
-			start:       []byte{0},
-			end:         nil, // Also tests start can be after end if end is nil
-			expectedErr: ErrDataInMissingRootProof,
-		},
-		{
-			name: "no roots in history and non-empty deleted keys",
-			proof: &ChangeProof{
-				HadRootsInHistory: false,
-				KeyChanges:        []KeyChange{{Key: []byte{1}}},
-			},
-			start:       nil,
-			end:         nil,
-			expectedErr: ErrDataInMissingRootProof,
-		},
-		{
-			name: "no roots in history and non-empty start proof",
-			proof: &ChangeProof{
-				HadRootsInHistory: false,
-				StartProof:        []ProofNode{{}},
-			},
-			start:       nil,
-			end:         nil,
-			expectedErr: ErrDataInMissingRootProof,
-		},
-		{
-			name: "no roots in history and non-empty end proof",
-			proof: &ChangeProof{
-				HadRootsInHistory: false,
-				EndProof:          []ProofNode{{}},
-			},
-			start:       nil,
-			end:         nil,
-			expectedErr: ErrDataInMissingRootProof,
-		},
-		{
-			name: "no roots in history; empty",
-			proof: &ChangeProof{
-				HadRootsInHistory: false,
-			},
-			start:       nil,
-			end:         nil,
-			expectedErr: nil,
-		},
-		{
-			name: "root in history; empty",
-			proof: &ChangeProof{
-				HadRootsInHistory: true,
-			},
```

Comment:
Removed tests where `HadRootsInHistory` is false because change proofs can no longer indicate that the server didn't have enough history to serve the change proof (a range proof is returned instead.)